### PR TITLE
Enable direct tracking from redmine

### PR DESCRIPTION
**Pull request by lukaselmer**
Enables timer start and stop directly from redmine (if you have the chrome toggl extension installed: https://chrome.google.com/webstore/detail/toggl-button-productivity/oejgccbfbmkkpaidnkphaiaecficdnfn?hl=en)

image

image

image

### CODE REVIEW

File: lib/assigner.rb [hunk 1]
```diff
@@ -60,4 +75,4 @@ def valid?
   def assignable?
     !issue_ids.empty?
   end
-end
\ No newline at end of file
+end
```
Comment (lukaselmer) [inline lib/assigner.rb:74]: 😱

**Review by lukaselmer — COMMENTED** [review]
self review

File: lib/assigner.rb [hunk 2]
```diff
@@ -26,13 +26,24 @@ def extract_toggle_id
   end
 
   def try_extract_id
-    id, text = self.comment.split(' ', 2)
-    success = id.to_i.to_s == id
-    if success
-      self.issue_ids << id.to_i
-      self.comment = text.to_s
-    end
-    return success
```
Comment (lukaselmer) [inline lib/assigner.rb:35]: why would my younger self code something like that!? 😲

File: spec/assigner_spec.rb [hunk 3]
```diff
@@ -40,6 +40,23 @@
     a.issue_ids.count.should == 1
   end
 
+  it 'should assign comment with ticket when toggled from redmine' do
```
Comment (rbuetzer) [inline spec/assigner_spec.rb:43]: Consider wrapping these `it` blocks inside a `context` to get rid of the "when toggled from redmine" part in the description.
http://betterspecs.org/#contexts

Additional benefit: If one day, we decide to drop Redmine, we could just delete this `context` block as it groups tests for a feature we do not need anymore.

File: spec/assigner_spec.rb [hunk 4]
```diff
@@ -40,6 +40,23 @@
...
+    a = Assigner.new '92345678: Feature #7682 Implement toggl directly from redmine'
+    a.issue_ids.first.should == 7682
+    a.issue_ids.count.should == 1
```
Comment (rbuetzer) [inline spec/assigner_spec.rb:46]: I assume, you have reasons for using should?
http://betterspecs.org/#should

**Review by rbuetzer — APPROVED** [review]

File: spec/assigner_spec.rb (hunk 4 above)
Comment (lukaselmer) [reply spec/assigner_spec.rb:46]: Haha, yes and no - it is just the old syntax and i do not want to rewrite the whole code. Absolutely agree with you considering the context parts, but same reason there.

If you feel like you can rewrite / restructure the tests in your 20% time :wink:

Comment (lukaselmer) [reply spec/assigner_spec.rb:46]: And thx for the review!

**Pull request merged**